cmd/sgx_epchook/main.go

```diff
@@ -30,45 +32,82 @@ import (
 )
 
 const (
-	namespace  = "sgx.intel.com"
-	epc        = "epc"
-	pathPrefix = "/status/capacity"
+	namespace = "sgx.intel.com"
+	epc       = "epc"
 )
 
-type patchExtendedResource struct {
-	Op    string `json:"op"`
-	Path  string `json:"path"`
-	Value uint64 `json:"value"`
+type patchNodeOp struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
 }
 
 func main() {
-	var register, affirm bool
+	var register, affirm, label, daemon bool
 	flag.BoolVar(&register, "register", false, "register EPC as extended resource")
 	flag.BoolVar(&affirm, "affirm", false, "return error if EPC is not available")
+	flag.BoolVar(&label, "node-label", false, "create node label")
+	flag.BoolVar(&daemon, "daemon", false, "run as a daemon")
 	flag.Parse()
 
+	klog.Infof("starting sgx_epchook")
+
 	// get the EPC size
 	var epcSize uint64
 	if cpuid.CPU.SGX.Available {
 		for _, s := range cpuid.CPU.SGX.EPCSections {
 			epcSize += s.EPCSize
 		}
 	}
+	klog.Infof("epc capacity: %d bytes", epcSize)
 
 	if epcSize == 0 && affirm {
 		klog.Fatal("SGX EPC is not available")
 	}
 
-	if register {
-		if err := registerExtendedResource(epcSize); err != nil {
-			klog.Fatal(err.Error())
-		}
-	} else {
+	if err := updateNode(epcSize, register, label); err != nil {
+		klog.Fatal(err.Error())
+	}
+
+	if !register {
 		fmt.Printf("%s/%s=%d", namespace, epc, epcSize)
```

Comment:
nit: change to `klog`?

Reply:
Do you mean to replace `fmt.Printf` with `klog`? Would that still work with NFD?

Reply:
ah true, sorry. scratch my comment